services/wfm/src/main/java/org/openkilda/wfm/topology/event/OfEventWfmTopology.java

```diff
@@ -91,13 +92,15 @@ public StormTopology createTopology() throws StreamNameCollisionException {
         //      just to pull out switchID.
         // (crimi) - not sure I agree here .. state can be maintained, albeit distributed.
         //
-        BoltDeclarer bd = builder.setBolt(DISCO_BOLT_ID, bolt, topologyConfig.getParallelism())
-                .shuffleGrouping(DISCO_SPOUT_ID);
-
         builder.setBolt(TOPO_ENG_BOLT_ID, createKafkaBolt(kafkaTopoEngTopic),
                 topologyConfig.getParallelism()).shuffleGrouping(DISCO_BOLT_ID, OfeLinkBolt.TOPO_ENG_STREAM);
         builder.setBolt(SPEAKER_BOLT_ID, createKafkaBolt(topologyConfig.getKafkaSpeakerTopic()),
                 topologyConfig.getParallelism()).shuffleGrouping(DISCO_BOLT_ID, OfeLinkBolt.SPEAKER_STREAM);
+        builder.setBolt(SPEAKER_DISCO_BOLT_ID, createKafkaBolt(topologyConfig.getKafkaSpeakerDiscoTopic()),
+                topologyConfig.getParallelism()).shuffleGrouping(DISCO_BOLT_ID, OfeLinkBolt.SPEAKER_DISCO_STREAM);
+
+        BoltDeclarer bd = builder.setBolt(DISCO_BOLT_ID, bolt, topologyConfig.getParallelism())
```

Comment:
Line 88
```
IStatefulBolt bolt = new OfeLinkBolt(topologyConfig);
```
can/should be moved here too.

PS Whole comment block that follow line 88 looks deprecated.